pt_miniscreen/pages/root/projects.py

```diff
@@ -0,0 +1,178 @@
+from pt_miniscreen.core.component import Component
+from pt_miniscreen.core.components import Image
+from pt_miniscreen.core.components.marquee_text import MarqueeText
+from pt_miniscreen.core.components.selectable_list import SelectableList
+from pt_miniscreen.core.utils import apply_layers, layer
+from pt_miniscreen.utils import get_image_file_path
+
+logger = logging.getLogger(__name__)
+
+
+class InvalidConfigFile(Exception):
+    pass
+
+
+class ProjectConfig:
+    CONFIG_FILE_SECTION = "project"
+
+    def __init__(
+        self, title: str, image: str, start: str, exit_condition: str, **kwargs
+    ) -> None:
+        self.title = title
+        if len(image) == 0 or not Path(image).is_file():
+            image = get_image_file_path("menu/settings.gif")
+        self.image = image
+        self.start = start
+        self.exit_condition = exit_condition
+
+    @classmethod
+    def from_file(cls, file):
+        config = configparser.ConfigParser()
+        try:
+            config.read(file)
+            project_config = config[cls.CONFIG_FILE_SECTION]
+
+            return ProjectConfig(
+                title=project_config["title"],
+                image=project_config.get("image", ""),
+                start=project_config["start"],
+                exit_condition=project_config.get("exit_condition"),
+            )
+        except Exception as e:
+            logger.warning(f"Error parsing file '{file}': {e}")
+            raise InvalidConfigFile(e)
+
+
+class ProjectPage(Component):
+    def __init__(self, project_config: ProjectConfig, **kwargs):
+        self.project_config = project_config
+        super().__init__(**kwargs)
+        self.text = self.create_child(
+            MarqueeText,
+            text=self.project_config.title,
+            font_size=14,
+            align="center",
+            vertical_align="center",
+        )
+        self.image = self.create_child(
+            Image,
+            image_path=self.project_config.image,
+        )
+
+    def render(self, image):
+        return apply_layers(
+            image,
+            [
+                layer(
+                    self.text.render,
+                    size=(120, 25),
+                    pos=(28, 0),
+                ),
+                layer(
+                    self.image.render,
+                    size=(25, 25),
+                    pos=(0, 0),
+                ),
+            ],
+        )
+
+
+class ProjectRow(Component):
+    def __init__(self, project_config: ProjectConfig, **kwargs) -> None:
+        self.project_config = project_config
+        super().__init__(**kwargs)
+        self.text = self.create_child(
+            MarqueeText,
+            text=self.project_config.title,
+            font_size=10,
+            align="center",
+            vertical_align="center",
+        )
+        self.page = partial(ProjectPage, self.project_config)
+
+    def render(self, image):
+        return apply_layers(
```

Comment:
I think this can return self.text.render, eg

```
def render(self, image)
  return self.text.render(image)
```

I don't remember the defaults for MarqueeText but I believe this will render left horizontally aligned and centre vertically aligned in the image passed, which will be the size of the row.

Reply:
good one! the image changed a bit (the text shifted 1 pixel based on the test snapshots) but I prefer the simplicity of the new render method